src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java

```diff
@@ -1401,6 +1360,85 @@ private boolean checkUnionErrors(RDBMS rdbms, HttpMessage msg, String response,
 		return false;
 	}
 
+	private void expressionBasedAttack(String param, String modifiedParamValue, String modifiedParamValueConfirm){
+		//those of you still paying attention will note that if handled as expressions (such as by a database), these represent the same value.
+		HttpMessage msg = getNewMsg();
+		setParameter(msg, param, modifiedParamValue);
+
+		try {
+			sendAndReceive(msg, false); //do not follow redirects
+		} catch (Exception ex) {
```

Comment:
Right, declaring that the method throws IOException allows to keep the previous behaviour.